frontend/util.lua

```diff
@@ -400,4 +400,102 @@ function util.splitToArray(str, splitter, capture_empty_entity)
     return result
 end
 
+--- Convert a Unicode codepoint (number) to UTF8 char
+--
+--- @int c Unicode codepoint
+--- @treturn string UTF8 char
+function util.unicodeCodepointToUtf8(c)
+    if c < 128 then
+        return string.char(c)
+    elseif c < 2048 then
+        return string.char(192 + c/64, 128 + c%64)
+    elseif c < 55296 or 57343 < c and c < 65536 then
+        return string.char(224 + c/4096, 128 + c/64%64, 128 + c%64)
+    elseif c < 1114112 then
+        return string.char(240 + c/262144, 128 + c/4096%64, 128 + c/64%64, 128 + c%64)
+    else
+        return util.unicodeCodepointToUtf8(65533) -- U+FFFD REPLACEMENT CHARACTER
+    end
+end
+
+local HTML_ENTITIES_TO_UTF8 = {
+    ["&lt;"] = "<",
+    ["&gt;"] = ">",
+    ["&amp;"] = "&",
+    ["&quot;"] = '"',
+    ["&apos;"] = "'",
+    ["&nbsp;"] = "\xC2\xA0",
+    ["&#(%d+);"] = function(x) return util.unicodeCodepointToUtf8( tonumber(x) ) end,
+    ["&#x(%x+);"] = function(x) return util.unicodeCodepointToUtf8( tonumber(x,16) ) end,
+}
+--- Replace HTML entities with their UTF8 equivalent in text
+--
+-- Supports only basic ones and those with numbers (no support
+-- for named entities like &eacute;)
+--- @int string text with HTML entities
+--- @treturn string UTF8 text
+function util.htmlEntitiesToUtf8(text)
+    for k,v in pairs(HTML_ENTITIES_TO_UTF8) do
+        text = text:gsub(k, v)
+    end
+    return text
+end
+
+--- Convert HTML to plain text
```

Comment:
OK, just seeing I forgot to update this comment:
`--- Remove HTML tags from text if text seems to be HTML
`
which I should have rewritten, like I did for the previous function's comment, to:
`--- Convert HTML to plain text if text seems to be HTML
`
(when I started, i actually was _removing HTML tags_ :) before I realized i needed to do conversion, and so the names and comments became about _converting HTML to plain text_, except this comment that I forgot, which may have led you on the _remove html_ track - sorry)